src/Formatter/NMEAFormatter.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2023 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "Units/Units.hpp"
+#include "NMEAFormatter.hpp"
+
+void
+FormatGPRMC(char *buffer, size_t size, const NMEAInfo &info) noexcept
+{
+  const GeoPoint location = info.location_available
+    ? info.location
+    : GeoPoint::Zero();
+
+  const BrokenDateTime now = info.time_available &&
+    info.date_time_utc.IsDatePlausible()
+    ? info.date_time_utc
+    : BrokenDateTime::NowUTC();
+
+  StringFormat(buffer, size,
+               "GPRMC,%02u%02u%02u,%c,%02u%02u.%03u,%c,%03u%02u.%03u,%c,%05.1f,%05.1f,%02u%02u%02u,%05.1f,%c",
+               now.hour, now.minute, now.second,
+               info.location_available ? 'A' : 'V',
+               (int)location.latitude.AbsoluteDegrees(),
```

Comment:
Does this suffer from rounding errors? Your way of implementing this looks suspicious - it uses different methods to extract degrees and the fractional part; to guarantee that it's free from rounding errors, only one method should be used. The IGC2NMEA code uses `Angle::ToDMS()` to extract all components.

Reply:
Will use `Angle::ToDMM()` as it has all necessary components ready to use.